components/user_settings/security/mfa_section/mfa_section.tsx

```diff
@@ -1,46 +1,50 @@
 // Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
 // See LICENSE.txt for license information.
 
-import PropTypes from 'prop-types';
 import React from 'react';
 import {FormattedMessage} from 'react-intl';
 
-import SettingItemMax from 'components/setting_item_max.jsx';
+import SettingItemMax from 'components/setting_item_max';
 import SettingItemMin from 'components/setting_item_min';
 import {browserHistory} from 'utils/browser_history';
 
 const SECTION_MFA = 'mfa';
 
-export default class MfaSection extends React.PureComponent {
-    static propTypes = {
-        active: PropTypes.bool.isRequired,
+type Props = {
+    active: boolean;
 
-        // Whether or not the current user has MFA enabled
-        mfaActive: PropTypes.bool.isRequired,
+    // Whether or not the current user has MFA enabled
+    mfaActive: boolean;
 
-        // Whether or not the current user can enable MFA based on their authentication type and the server's settings
-        mfaAvailable: PropTypes.bool.isRequired,
+    // Whether or not the current user can enable MFA based on their authentication type and the server's settings
+    mfaAvailable: boolean;
 
-        // Whether or not this server enforces that all users have MFA
-        mfaEnforced: PropTypes.bool.isRequired,
+    // Whether or not this server enforces that all users have MFA
+    mfaEnforced: boolean;
 
-        updateSection: PropTypes.func.isRequired,
-        actions: PropTypes.shape({
-            deactivateMfa: PropTypes.func.isRequired,
-        }),
-    };
+    updateSection: (section: string) => void;
+    actions: {deactivateMfa: () => any};
```

Comment:
Non-blocking, but this can return `void` since we aren't using the return value.

Reply:
we cannot return void since error is type any.
const {error} = await this.props.actions.deactivateMfa(); line 50

Reply:
I see the problem. deactivateMfa() should return a Promise with an an object with an object error, which might be undefined, and which has an attribute message. Would you mind checking the changes on the index.js in this PR and follow the same pattern?
You may (or may not) have to migrate the file where deactivateMfa is declared.

https://github.com/mattermost/mattermost-webapp/pull/4837/files